app/models/assignment.rb

```diff
@@ -1,13 +1,214 @@
+      # Raises error if the user is already a participant
+      raise "The user #{user.name} is already a participant."
+    end
+    # Create a new AssignmentParticipant associated with the assignment and user
+    new_part = AssignmentParticipant.create(assignment_id: self.id,
+                                            user_id: user.id)
+    # Set the participant's handle
+    new_part.set_handle
+    # Return the newly created AssignmentParticipant
+    new_part
+  end
+
+
+  # Remove a participant from the assignment based on the provided user_id.
+  # This method finds the AssignmentParticipant with the given assignment_id and user_id,
+  # and then deletes the corresponding record from the database.
+  # No return value; the participant is removed from the assignment.
+  def remove_participant(user_id)
+    # Find the AssignmentParticipant associated with this assignment and user
+    assignment_participant = AssignmentParticipant.where(assignment_id: self.id, user_id: user_id).first
+    # Delete the AssignmentParticipant record
+    if assignment_participant
+      assignment_participant.destroy
+    end
+  end
+
+  # Remove the assignment from the associated course.
+  # This method sets the course_id of the assignment to nil, effectively removing its course association.
+  # Returns the modified assignment object with course_id set to nil.
+  def remove_assignment_from_course
+    # Set the course_id of the assignment to nil
+    self.course_id = nil
+    # Return the modified assignment
+    self
+  end
+
+
+
+ # Assign a course to the assignment based on the provided course_id.
+  # If the assignment already belongs to the specified course, an error is raised.
+  # Returns the modified assignment object with the updated course assignment.
+  def assign_courses_to_assignment(course_id)
+    # Find the assignment by its ID
+    assignment = Assignment.where(id: id).first
+    # Check if the assignment already belongs to the provided course_id
+    if assignment.course_id == course_id
+      # Raises error if the assignment already belongs to the provided course_id
+      raise "The assignment already belongs to this course id."
+    end
+    # Update the assignment's course assignment
+    assignment.course_id = course_id
+    # Return the modified assignment
+    assignment
+  end
+
+
+  # Create a copy of the assignment, including its name, instructor, and course assignment.
+  # The new assignment is named "Copy of [original assignment name]".
+  # Returns the newly created assignment object, which is a copy of the original assignment.
+  def copy
+    copied_assignment = Assignment.new(
+        name: "Copy of #{self.name}",
+        course_id: self.course_id
+      )
+
+    # Assign the correct instructor to the copied assignment
+    copied_assignment.instructor = self.instructor
+
+    # Save the copied assignment to the database
+    copied_assignment.save
+
+    copied_assignment
+
+  end
+  def is_calibrated?
```

Comment:
The methods to get fields of assignment won't be needed if a show method is created that returns the assignment.